Python/errors.c

```diff
@@ -306,16 +306,7 @@ PyErr_NormalizeException(PyObject **exc, PyObject **val, PyObject **tb)
     }
     /* normalize recursively */
     tstate = PyThreadState_GET();
-    if (++tstate->recursion_depth > Py_GetRecursionLimit()) {
```

Comment:
If this is a dead code this issue couldn't exist. Adding `Py_FatalError()` here will made the interpreter crashing earlier in the original reported example.